dev-libs/dmalloc/dmalloc-5.6.5-r1.ebuild

```diff
@@ -0,0 +1,62 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit autotools
+
+DESCRIPTION="A Debug Malloc Library"
+HOMEPAGE="https://dmalloc.com"
+SRC_URI="https://dmalloc.com/releases/${P}.tgz"
+
+LICENSE="CC-BY-SA-3.0"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~mips ~ppc ~ppc64 ~riscv ~s390 ~sparc ~x86"
+IUSE="threads"
+
+BDEPEND="sys-apps/texinfo"
+
+PATCHES=(
+	"${FILESDIR}"/${PN}-5.6.5-add-destdir-support.patch
+	"${FILESDIR}"/${PN}-5.6.5-allow-overriding-ar-and-ld.patch
+	"${FILESDIR}"/${PN}-5.6.5-set-soname-version.patch
+	"${FILESDIR}"/${PN}-5.6.5-configure-c99.patch
+	"${FILESDIR}"/${PN}-5.6.5-fix-cxx-check.patch
+)
+
+src_prepare() {
+	default
+
+	eautoreconf
+}
+
+src_configure() {
+	econf \
+		--enable-cxx \
+		--enable-shlib \
+		$(use_enable threads)
+}
+
+src_test() {
+	# mv: cannot stat 'aout': No such file or directory
+	emake -j1 heavy
+}
+
+src_install() {
+	default
+
+	soname_link() {
```

Comment:
Double check the order is right wrt f67f7c6453fdd128489ea675935302099d66c1f5?

Reply:
Didn't know about this and was just trying to keep as close to original behaviour as possible.

Following that order would simplify the patches.

Reply:
Never mind, got the symbolic link directions the wrong around. It wont simplify it.

But the order is still wrong in the ebuild.

Reply:
```
$ qlist dmalloc -v
dev-libs/dmalloc-5.6.5-r1 CONTENTS:
/usr/share/doc/dmalloc-5.6.5-r1/TODO.bz2
/usr/share/doc/dmalloc-5.6.5-r1/README.md.bz2
/usr/bin/dmalloc
/usr/lib64/libdmallocxx.so -> libdmallocxx.so.5
/usr/lib64/libdmallocxx.so.5 -> libdmallocxx.so.5.6.5
/usr/lib64/libdmalloc.so -> libdmalloc.so.5
/usr/lib64/libdmalloc.so.5 -> libdmalloc.so.5.6.5
/usr/lib64/libdmalloc.so.5.6.5
/usr/lib64/libdmallocthcxx.so.5.6.5
/usr/lib64/libdmallocth.so.5.6.5
/usr/lib64/libdmallocxx.so.5.6.5
/usr/include/dmalloc.h
/usr/lib64/libdmallocth.so.5 -> libdmallocth.so.5.6.5
/usr/lib64/libdmallocth.so -> libdmallocth.so.5
/usr/lib64/libdmallocthcxx.so.5 -> libdmallocthcxx.so.5.6.5
/usr/lib64/libdmallocthcxx.so -> libdmallocthcxx.so.5
```